arborist/server.go

```diff
@@ -280,27 +280,66 @@ func (server *Server) handleAuthMappingPOST(w http.ResponseWriter, r *http.Reque
 		Username string `json:"username"`
 		ClientID string  `json:"clientID"`
 	}{}
-	err := json.Unmarshal(body, &requestBody)
-	if err != nil {
-		msg := fmt.Sprintf("could not parse JSON: %s", err.Error())
-		server.logger.Info("tried to handle auth mapping request but input was invalid: %s", msg)
-		errResponse = newErrorResponse(msg, 400, nil)
+
+	// Try to get username or clientID from the JWT.
+	username := ""
+	clientID := ""
+	if authHeader := r.Header.Get("Authorization"); authHeader != "" {
+		server.logger.Info("Attempting to get username or clientID from jwt...")
+
+		userJWT := strings.TrimPrefix(authHeader, "Bearer ")
+		userJWT = strings.TrimPrefix(userJWT, "bearer ")
+		scopes := []string{"openid"}
+		info, err := server.decodeToken(userJWT, scopes)
+		if err != nil {
+			// Return 400 on failure to decode JWT
+			msg := fmt.Sprintf("tried to get username/clientID from jwt, but jwt decode failed: %s", err.Error())
+			server.logger.Info(msg)
+			errResponse = newErrorResponse(msg, 400, nil)
+		} else {
+			username = info.username
+			clientID = info.clientID
```

Comment:
related to my previous comment about the `if (username == "") && (clientID == "")` block: i think we should add `if (username == "") && (clientID == "") { error invalid jwt for this endpoint }` here